doc/website/blog/serverless-graphql-using-aws.md

```diff
@@ -0,0 +1,556 @@
+
+First, we create an empty directory to which we change:
+```console
+mkdir entgo-aws-appsync && cd $_
+```
+Second, we enable go modules and install the Ent toolset:
+```console
+go mod init entgo-aws-appsync
+go mod tidy
+go get -d entgo.io/ent/cmd/ent
+```
+Third, we create the `Todo` schema
+```console
+go run entgo.io/ent/cmd/ent init Todo
+```
+and add the `title` field:
+```go title="ent/schema/todo.go"
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/field"
+)
+
+// Todo holds the schema definition for the Todo entity.
+type Todo struct {
+	ent.Schema
+}
+
+// Fields of the Todo.
+func (Todo) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title"),
+	}
+}
+
+// Edges of the Todo.
+func (Todo) Edges() []ent.Edge {
+	return nil
+}
+```
+Finally, we perform the Ent code generation:
+```console
+go generate ./ent
+```
+
+Using the Ent, we write a set of resolver functions, which implement the create, read, and delete operations on the todos:
+```go title="internal/handler/resolver.go"
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"entgo-aws-appsync/ent"
+	"entgo-aws-appsync/ent/todo"
+)
+
+type TodosInput struct{}
+
+func Todos(ctx context.Context, client *ent.Client, input TodosInput) ([]*ent.Todo, error) {
+	return client.Todo.
+		Query().
+		All(ctx)
+}
+
+type TodoByIDInput struct {
+	ID string `json:"id"`
+}
+
+func TodoByID(ctx context.Context, client *ent.Client, input TodoByIDInput) (*ent.Todo, error) {
+	tid, err := strconv.Atoi(input.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing todo id: %w", err)
+	}
+	return client.Todo.
+		Query().
+		Where(todo.ID(tid)).
+		Only(ctx)
+}
+
+type AddTodoInput struct {
+	Title string `json:"title"`
+}
+
+type AddTodoOutput struct {
+	Todo *ent.Todo `json:"todo"`
+}
+
+func AddTodo(ctx context.Context, client *ent.Client, input AddTodoInput) (*AddTodoOutput, error) {
+	t, err := client.Todo.
+		Create().
+		SetTitle(input.Title).
+		Save(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating todo: %w", err)
+	}
+	return &AddTodoOutput{Todo: t}, nil
+}
+
+type RemoveTodoInput struct {
+	TodoID string `json:"todoId"`
+}
+
+type RemoveTodoOutput struct {
+	Todo *ent.Todo `json:"todo"`
+}
+
+func RemoveTodo(ctx context.Context, client *ent.Client, input RemoveTodoInput) (*RemoveTodoOutput, error) {
+	t, err := TodoByID(ctx, client, TodoByIDInput{ID: input.TodoID})
+	if err != nil {
+		return nil, fmt.Errorf("failed querying todo with id %q: %w", input.TodoID, err)
+	}
+	err = client.Todo.
+		DeleteOne(t).
+		Exec(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed deleting todo with id %q: %w", input.TodoID, err)
+	}
+	return &RemoveTodoOutput{Todo: t}, nil
+}
+```
+Using input structs for the resolver functions allows mapping the GraphQL request arguments.
+Using output structs allows us to return multiple objects for more complex operations.
```

Comment:
```suggestion
Using output structs allows for us to return multiple objects for more complex operations.
```

Reply:
Are you sure about this one?

Reply:
Yes. `Allow` is used in a context where permission is given. `Allow for` is used if something is enabled because if some preceding action. 

https://english.stackexchange.com/questions/30069/whats-the-difference-between-allow-and-allow-for